cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/AuthorizationResult.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+/**
+ * A helper class to support the response for different method calls that returns an object but also needs to
+ * get the {@link AuthorizationResponse} for the call.
+ *
+ * @param <T> : This is the return object of the method call
+ */
+public class AuthorizationResult<T> {
```

Comment:
nit: name it AuthorizedResult. It's reflecting the purpose better as it's not a result of an authorization operation, it's a result of some operation that was authorized